### ISSUE TITLE
Consume Git configuration variables to pre-fill the default remote to push branches and tags to

### ISSUE THREAD

**Issue by MetRonnie** [mhutchie/vscode-git-graph#424]
**Describe the improvement that you'd like**

When you right click on a branch and choose `Push branch...` from the context menu, the remote to push to in the dropdown always defaults to `origin` for me, even for a branch `foo` that has `branch.foo.pushremote = upstream` set in `.git/config`.

I'd like it if it read the `branch.foo.pushremote` value and chose that remote instead.

**Additional context (optional)**

- https://git-scm.com/docs/git-config#Documentation/git-config.txt-branchltnamegtpushRemote

  > When on branch `<name>`, it overrides `branch.<name>.remote` for pushing. It also overrides `remote.pushDefault` for pushing from branch `<name>`. When you pull from one place (e.g. your upstream) and push to another place (e.g. your own publishing repository), you would want to set `remote.pushDefault` to specify the remote to push to for all branches, and use this option to override it for a specific branch
- I don't know if vscode-git-graph already obeys `remote.pushdefault` from the config or not
- My `.git/config` for the repo in question includes:

  ```ini
    [remote "upstream"]
        url = something
        fetch = +refs/heads/*:refs/remotes/upstream/*
    [remote "origin"]
        url = something_else
        fetch = +refs/heads/*:refs/remotes/origin/*
    [remote]
        pushdefault = origin
    [branch "foo"]
        remote = upstream
        pushremote = upstream
        merge = refs/heads/v1
    [branch "master"]
        remote = upstream
        merge = refs/heads/master
  ```

**Comment by mhutchie** [comment]
Hi @MetRonnie,

Thanks for raising this improvement request! I wasn't aware of `remote.pushDefault` / `branch.*.pushDefault`, so thank you for making me aware of it, linking to the Git documentation of it, and providing a clear example of how you use it for your repository.

Currently (when there are multiple remotes for the repository) the "Push Branch" dialog simply lists all remotes, and defaults to the `origin` remote if it exists. It doesn't read `remote.pushDefault` or `branch.*.pushDefault` at all - the only place where the Git config is currently being read by Git Graph is for the Repository Settings Widget.

I definitely agree that the "Push Branch" dialog should default to `remote.pushDefault` / `branch.*.pushDefault` if defined. I'll include this in an upcoming release.

**Comment by mhutchie** [comment]
This will be available in v1.29.0.

I took the opportunity to completely overhaul Git configuration fetching & usage across Git Graph, to enable this feature and be better positioned for utilising other Git configuration values throughout Git Graph in the future.

If you'd like to use it before the next release, you can download [v1.29.0-beta.2](https://github.com/mhutchie/vscode-git-graph/releases/tag/v1.29.0-beta.2), and install it following the instructions provided [here](https://github.com/mhutchie/vscode-git-graph/wiki/Downloading-&-Installing-Releases#manual-installation).

**Comment by MetRonnie** [comment]
Thank you, and can I take the opportunity to say what a great extension this is!
